packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -753,6 +753,46 @@ describe('TransactionController', () => {
 
         expect(transactions[0].status).toBe(TransactionStatus.submitted);
       });
+
+      it('only reads the current chain id once to filter for approved transactions', async () => {
```

Comment:
Not sure if we'll keep this test or try to find a better way to test this, but if we do keep it can you explain the flow by which instantiating the class leads to the filtering for approved transactions?

Reply:
Oh I see, its in `onBootCleanUp` -> `submitApprovedTransactions` -> `getCurrentChainTransactionsByStatus`.

Reply:
So it seems like our goal with this flow would to remove the filtering by chainId altogether no? So then we just filter by approved status and submit anything that's approved regardless of whether it matches the globally selected network.

Reply:
Created a ticket for this here: https://app.zenhub.com/workspaces/devex-63bee08a4e3b9d001108416e/issues/gh/metamask/metamask-planning/1893